Change minimum IDF version for `cbor` component to IDF v5.0

dhavalgujar opened the pull request:
# Checklist

- [x] Component contains License
- [x] Component contains README.md
- [x] Component contains idf_component.yml file with `url` field defined
- [x] Component was added to [upload job](https://github.com/espressif/idf-extra-components/blob/master/.github/workflows/upload_component.yml#L18)
- [x] Component was added to [build job](https://github.com/espressif/idf-extra-components/blob/master/test_app/CMakeLists.txt#L8)
- [] _Optional:_ Component contains unit tests
- [x] CI passing

# Change description
- IDF versions below v5.0 already include the `cbor` component in source format.
- Due to this, there is a conflict if the `cbor` component is used on lower versions of IDF and the only way to fix it is by adding a rule to the `idf_component.yml`
- This commit sets the minimum version to IDF v5.0 to fix this behaviour.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/espressif/idf-extra-components?pullRequest=122) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/espressif/idf-extra-components?pullRequest=122) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/espressif/idf-extra-components?pullRequest=122) it.</sub>

dhavalgujar commented:
> I think we still want to have the option of installing the latest version of cbor component in IDF v4.x.

Ah, I see. I thought the version of `cbor` in IDF v4.4 and v0.6.0 from the IDF Component Registry were the same.

> Could you explain why this is problematic?

It isn't problematic, it is just a different pattern than what has been followed by other components which were also removed from IDF in v5.0 and added to the IDF Component Registry.
For example: The `led_strip` component exists as [a part of](https://github.com/espressif/esp-idf/tree/release/v4.4/examples/common_components/led_strip) IDF v4.4 but the latest one from the [registry](https://components.espressif.com/components/espressif/led_strip) has IDF v5.0 as the minimum. 
Again, there might be other reasons why that component lists IDF v5.0 as the minimum which I am unaware of.

> I am not sure whether your proposed change will be effective.

Right, this might not work unless we can change the minimum IDF compatibility of an older version which I'm assuming isn't possible and probably a bad idea?

mahavirj commented:
Lets keep current IDF 4.3 version approach for `cbor` component. Provides an advantage of easy upgrade across previous releases..
